phoenicis-scripts/src/main/java/org/phoenicis/scripts/engine/injectors/IncludeInjector.java

```diff
@@ -20,20 +20,22 @@ public IncludeInjector(ScriptFetcher scriptFetcher) {
 
     @Override
     public void injectInto(PhoenicisScriptEngine phoenicisScriptEngine) {
-        final Set<String> includedScripts = new HashSet<>();
+        // store included scripts (include path -> JS object)
+        final Map<String, Object> includedScripts = new HashMap<>();
```

Comment:
Maybe also add a check for circular dependencies (i.e. `A` -> `B` -> `A`)?

This should be relatively easy to implement with a `Stack`

Reply:
Can we do that later? It doesn't really belong to the issue which is fixed in this PR.

Reply:
As long as we don't forget it I'm fine with that. I think this is an important check because it makes it easier to find certain types of errors.

Reply:
See https://github.com/PhoenicisOrg/phoenicis/issues/2048.